showkase-screenshot-testing-shot/src/main/java/com.airbnb.android.showkase.screenshot.testing.shot/ShotShowkaseScreenshotTest.kt

```diff
@@ -0,0 +1,20 @@
+package com.airbnb.android.showkase.screenshot.testing.shot
+
+import android.graphics.Bitmap
+import com.airbnb.android.showkase.screenshot.testing.ShowkaseScreenshotTest
+import com.airbnb.android.showkase.screenshot.testing.ShowkaseScreenshotType
+import com.karumi.shot.ScreenshotTest
+
+abstract class ShotShowkaseScreenshotTest: ShowkaseScreenshotTest, ScreenshotTest {
+
+    override fun onScreenshot(
+        id: String,
+        name: String,
+        group: String,
+        styleName: String?,
+        screenshotType: ShowkaseScreenshotType,
+        screenshotBitmap: Bitmap
+    ) {
+        compareScreenshot(composeTestRule)
```

Comment:
* `compareScreenshot` also takes a `name` as a parameter, we could pass a concatenation of `name/group/styleName`.
* Also has an overload taking a `Bitmap` instead of the `composeTestRule`

Reply:
Another note, I did a similar integration manually on a device running api 31, and I got the following error regarding permissions:

```
Execute ExampleShowkaseScreenshotTest_ShowkaseCodegen.composable_screenshot_test_2: FAILED
junit.framework.AssertionFailedError: junit.framework.AssertionFailedError: Failed to grant permissions, see logcat for details
```

Leaving `onScreenshot` empty still causes the error, so I guess this is caused by Showkase. Maybe worth opening an issue.
Shot seems to be handling this well, even on Android 11+ where `WRITE_EXTERNAL_STORAGE` is deprecated.

If anyone has the same issue, in the mean time can be fixed by overloading `takeComposableScreenshot` functions to bypass showkase screenshot process:

```kotlin
@ShowkaseScreenshot(rootShowkaseClass = RootModule::class)
open class ExampleShowkaseScreenshotTest : ShowkaseScreenshotTest, ScreenshotTest {
    @get:Rule
    override val composeTestRule = createComposeRule()

    // Do the same for typo/color functions
    override fun takeComposableScreenshot(showkaseBrowserComponent: ShowkaseBrowserComponent) {
        // Disable animations for screenshots to make them deterministic
        composeTestRule.mainClock.autoAdvance = false
        composeTestRule.setContent { showkaseBrowserComponent.component() }
        compareScreenshot(
            composeTestRule,
            name = showkaseBrowserComponent.run { "${group}_${componentName}_${styleName}" }
        )
    }

    override fun onScreenshot(
        id: String,
        name: String,
        group: String,
        styleName: String?,
        screenshotType: ShowkaseScreenshotType,
        screenshotBitmap: Bitmap
    ) {
        // no-op
    }
}
```

Reply:
The `ShowkaseScreenshotTest` changes in this branch https://github.com/marcorighini/Showkase/commit/ee3176577b357cdbacb832ab4e6a2e93e3a5f6f4#diff-eefe4c579476fb035be5abc91fcb8701101f66f9b9cd494e774f4f89d83ccf20 would be simplify this, and we wouldn't need to override the 3 functions to bypass current screenshot mechanism

Reply:
Maybe @marcorighini can contribute those changes back into this and we could also create a `showkase-screenshot-testing-paparazzi` module? Thoughts @vinaygaba?

Reply:
@badoualy for the error that you were seeing, I added this in the Manifest and that seems to solve the problem when I tested it - https://github.com/airbnb/Showkase/pull/232/files#diff-c53f7bc57c4a7ffb92824cc2598ff253042b28a0d38f69eead012e62f7c41936R9

@pmecho As for Paparazzi, my follow up PR is going to be the `showkase-paparazzi` artifact as I'm very excited about it. It will dramatically simplify things as we can simply rely on parametrized testing instead of leveraging code generation. The reason I had to use code generation in the first place was that Firebase Test Labs did not work well with parameterized tests due to something to do with sharding iirc. 

The pararazzi implementation will look very similar to what Alex describes in this comment - https://github.com/airbnb/Showkase/issues/207#issuecomment-1114042072

Reply:
@vinaygaba `requestLegacyExternalStorage` no longer works when targeting api 30+
https://developer.android.com/training/data-storage/use-cases#opt-out-in-production-app